include/rmm/detail/runtime_async_alloc.hpp

```diff
@@ -70,6 +75,28 @@ struct runtime_async_alloc {
     }
     return (supported_handle_types_bitmask & handle_type) == handle_type;
   }
+
+  /**
+   * @brief Check whether `cudaMemPoolCreateUsageHwDecompress` is a supported
+   * pool property on the present CUDA driver version.
+   *
+   * Requires RMM to be built with a supported CUDA version 12.8+, otherwise
+   * this always returns false.
+   *
+   * @return true if supported
+   * @return false if unsupported
+   */
+  static bool is_hwdecompress_supported()
+  {
+#if defined(CUDA_VERSION) && CUDA_VERSION >= RMM_MIN_HWDECOMPRESS_CUDA_DRIVER_VERSION
+    // Check if hardware decompression is supported (requires CUDA 12.8 driver or higher)
+    int driver_version{};
+    RMM_CUDA_TRY(cudaDriverGetVersion(&driver_version));
+    return driver_version >= RMM_MIN_HWDECOMPRESS_CUDA_DRIVER_VERSION;
```

Comment:
*suggestion (nonblocking)*: only do this once?

```suggestion
    static bool is_supported = []() {
        int version{};
        RMM_CUDA_TRY(cudaDriverGetVersion(&version));
        return version >= RMM_MIN_HWDECOMPRESS_CUDA_DRIVER_VERSION;
    }();
    return is_supported;
```

Reply:
Good call. We can cache this.